src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -229,6 +240,14 @@ internal GeneratorDriverState RunGeneratorsCore(Compilation compilation, Diagnos
                             ex = e;
                         }
 
+                        // if the pipeline registered any syntax callbacks, create a receiver to handle them
+                        if (sourcesBuilder.SyntaxTransformNodes.Count > 0)
```

Comment:
>if (sourcesBuilder.SyntaxTransformNodes.Count > 0) [](http://example.com/codeflow?start=24&length=50)

Is this `if` check necessary, or could we simply execute the `if` body unconditionally?